Dockerfile

```diff
@@ -1,31 +1,23 @@
-FROM centos:7
-
-RUN yum -y update && \
-    yum -y install java-1.8.0-openjdk
-
-ENV JAVA_HOME /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.292.b10-1.el7_9.x86_64
-
-ENV JRE_HOME /usr/lib/jvm/jre
+FROM 345280441424.dkr.ecr.ap-south-1.amazonaws.com/ark_base_java8:latest
 
 RUN useradd -r -g root tomcat
 
-
-ADD  http://apache.cs.utah.edu/tomcat/tomcat-9/v9.0.46/bin/apache-tomcat-9.0.46.tar.gz /tmp
+ADD  https://archive.apache.org/dist/tomcat/tomcat-9/v9.0.22/bin/apache-tomcat-9.0.22.tar.gz /tmp
 
 RUN cd /tmp && \
-    tar -xf apache-tomcat-9.0.46.tar.gz && \
-    mv apache-tomcat-9.0.46 /opt/tomcat/ && \
+    tar -xf apache-tomcat-9.0.22.tar.gz && \
+    rm apache-tomcat-9.0.22.tar.gz && \
+    mv apache-tomcat-9.0.22 /opt/tomcat/ && \
     chown -R tomcat: /opt/tomcat && \
```

Comment:
The `tomcat` user should not own anything more than strictly required (principle of least privilege). By default, all the tomcat-related files should be owned by root and not writable by the `tomcat` user. Tomcat is a web server, so I guess the only thing it need to write to is a temporary directory for cached files (but I don't know much about Tomcat...) In any case, only the files and directories should be owned by the `tomcat` user if doing otherwise would prevent the Tomcat server from performing properly.

Reply:
`tomcat` user requires read/write and execute permissions sometimes for logging to files,logrorate, and running the java applications within the tomcat user working directory. Here it is `/opt/tomcat` . Here are the tomcat user permissions 
```
bash-4.2$ cd /opt/tomcat
bash-4.2$ ls -lh
total 144K
-rw-r----- 1 tomcat root  19K Jul  4  2019 BUILDING.txt 
-rw-r----- 1 tomcat root 5.3K Jul  4  2019 CONTRIBUTING.md
-rw-r----- 1 tomcat root  56K Jul  4  2019 LICENSE
-rw-r----- 1 tomcat root 2.3K Jul  4  2019 NOTICE
-rw-r----- 1 tomcat root 3.2K Jul  4  2019 README.md
-rw-r----- 1 tomcat root 6.7K Jul  4  2019 RELEASE-NOTES
-rw-r----- 1 tomcat root  16K Jul  4  2019 RUNNING.txt
drwxr-x--- 2 tomcat root 4.0K Jul  8 16:14 bin
drwx------ 2 tomcat root 4.0K Jul  4  2019 conf
drwxr-x--- 2 tomcat root 4.0K Jul  8 16:14 lib
drwxr-x--- 2 tomcat root 4.0K Jul  4  2019 logs
drwxr-x--- 2 tomcat root 4.0K Jul  8 16:14 temp
drwxr-x--- 7 tomcat root 4.0K Jul  4  2019 webapps
drwxr-x--- 2 tomcat root 4.0K Jul  4  2019 work

```

Reply:
@naveen-armedia  Fabrice was saying to narrow it down to specifically what tomcat requires.  I doubt it is going to be reading/modifying RELEASE-NOTES, README.md, etc.

Reply:
All logging should go to stdout. As far as I know, Java does not need write permissions to a jar/war file to run it, is that correct?

Reply:
No, it requires read, write and execute permissions. `tomcat` unpacks jar/war files to the new directory that contains the classes when deployed to the `webapps` directory.   All the logging configuration is done in either context.xml or logging.properties or in log4j.property config files and will take care of keeping logs to stdout in the application images that inherit this image.

Reply:
@naveen-armedia Wherever Tomcat requires write access, you can provide it so. I am quite sure that Tomcat does not require write access for all files and directories under `/opt/tomcat`. For example the `README.md` won't be touched by Tomcat. This is a trivial example but hopefully you understand my point. 

Please change the ownership of all files and directories under `/opt/tomcat` to `root:root`, and give write access to the `tomcat` user only for those files and directories that Tomcat absolutely need to write to. Thanks.